xarray/core/common.py

```diff
@@ -576,7 +576,13 @@ def where(self, cond, other=None, drop=False):
             outobj = self
             outcond = cond
 
-        return outobj._where(outcond)
+        # preserve attributes
+        out = outobj._where(outcond)
+        out.attrs = self.attrs
+        if isinstance(out, Dataset):
```

Comment:
Yes, good idea. Should the Dataset version check for the compatibility of ``other``, or should we just assume that ``other`` will always have the same variables?